### PR TITLE
Remove IFluidHTMLView, ReactViewAdapter, HTMLViewAdapter

### PR DESCRIPTION
**Pull request by ChumpChief**
I deprecated these previously (except `ReactViewAdapter` which I'm deprecating in #14189), and so this change removes them in `next`.

In the process I found that neither the view-interfaces nor view-adapters packages were actually generating their typetests, so added those relevant scripts.  The test build/tsconfig also seemed non-standard, so I tried to bring that in line with what other packages are doing.

With my recent change to SMDE (#14087) webpack-fluid-loader can safely expect to get an `IFluidMountableView` from all of our demos, so the fallback to `HTMLViewAdapter` isn't necessary anymore.

### CODE REVIEW

File: BREAKING.md
Comment (ssimic2) [inline BREAKING.md:33]: Maybe just call out the specific release where we removed these APIs, so it's clear if it's minor/major etc. (As a general rule we try to aim majors, so its good to make the "when" visible here)

**Comment by ssimic2** [comment]
Nice to see us extending efforts to move out the *views* logic. 

What about MountableView? One ask here is to communicate the overall vision/direction, and some of the key principles behind the direction. It could be in Readme at this point (stating the direction).

**Review by ssimic2 — APPROVED** [review]

**Comment by ChumpChief** [comment]
> Nice to see us extending efforts to move out the _views_ logic.
> 
> What about MountableView? One ask here is to communicate the overall vision/direction, and some of the key principles behind the direction. It could be in Readme at this point (stating the direction).

This removal is actually the completion of a very old project I've had on the back burner for around 3 years (see #1041) rather than purely about the API surface reduction - we may still want to do something with MountableView but it will depend on what we want the future of the container-views pattern to be.  I'll be sure to think about this in the context of the API surface reduction project.

File: BREAKING.md
Comment (ChumpChief) [reply BREAKING.md]: Done in latest

**Comment by ChumpChief** [comment]
Actually this is not yet ready to merge - I'll wait for #14189 to port to `next` first to simplify merge resolution.

**Comment by ChumpChief** [comment]
This is now ready to merge, if someone with merge superpowers can do for me :)

**Pull request merged**
